posydon/utils/common_functions.py

```diff
@@ -218,26 +218,40 @@ def roche_lobe_radius(m1, m2, a_orb=1):
     """
     
     ## catching if a_orb is an empty array or is an array with invalid separation values
-    if isinstance(a_orb, np.ndarray) and (not np.any(a_orb) or np.any(a_orb <=0)):
-        Pwarn("Trying to compute RL radius for binary with no separation", "EvolutionWarning")
-        a_orb = np.full_like(a_orb, np.nan)
+    if isinstance(a_orb, np.ndarray):
+        ## if array is empty (or all zeros), fill with NaN values
+        if a_orb.size == 0:
+            Pwarn("Trying to compute RL radius for binary with invalid separation", "EvolutionWarning")
+            a_orb = np.full_like(a_orb, np.nan, dtype=np.float64)
```

Comment:
This isn't doing anything, because an empty ndarray stays empty. If you like to get np.nan values in there you'd need to change the shape to make all dimensions with 0 shape to become a shape 1.
e.g. `a_orb = np.full([1 if s==0 else s for s in a_orb.shape], np.nan, dtype=np.float64)`

Reply:
If the shape is zero the size will not be zero, so the fix I have implemented should work.

Reply:
But here we are behind a `a_orb.size == 0`, thus the ndarray is empty.

Reply:
Please see the changes that have been made in the code before commenting on threads that have been resolved.